.github/workflows/check.yml

```diff
@@ -0,0 +1,21 @@
+name: Check
+on:
+  [pull_request]
+jobs:
+    unit-tests:
+        name: Tests
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout Repository
+              uses: actions/checkout@v2
+
+            - name: Setup Java 15
+              uses: actions/setup-java@v1
+              with:
+                  java-version: '15'
```

Comment:
If you already adjusted the project to Gradle 7.3 (and solved all deprecation warnings in functional tests), there is a chance to make it run with JDK 17 (PIT itself supports it).

I cannot easily switch to Gradle 7.x due to my Continuous Delivery mechanism which uses the old maven plugin (removed in 7.x).